workflows/data_pipelines/spectacle/spectacle_processor.py

```diff
@@ -0,0 +1,53 @@
+from dag_datalake_sirene.helpers import DataProcessor, Notification
+
+
+class SpectacleProcessor(DataProcessor):
+    def __init__(self):
+        from dag_datalake_sirene.workflows.data_pipelines.spectacle.spectacle_config import (
+            SPECTACLE_CONFIG,
+        )
+
+        super().__init__(SPECTACLE_CONFIG)
+
+    def preprocess_data(self):
+        import pandas as pd
+        from airflow.operators.python import get_current_context
+
+        df_spectacle = pd.read_csv(
+            self.config.file_download_destination, dtype=str, sep=";"
+        )
+
+        df_spectacle = (
+            df_spectacle.assign(
+                statut_du_recepisse=lambda x: x["Statut du récépissé"],
+                est_entrepreneur_spectacle=True,
+                siren=lambda x: x[
+                    "SIRET (personne morale) / SIREN (personne physique)"
+                ].str[:9],
+            )
+            .loc[
+                lambda x: x["siren"].notna() & x["siren"].str.isdigit(),
+                ["siren", "statut_du_recepisse", "est_entrepreneur_spectacle"],
+            ]
+            .groupby(["siren", "est_entrepreneur_spectacle"])
+            # If at least one of `statut` values is valid, then the value we keep is `valide
+            .agg(
+                statut_entrepreneur_spectacle=(
+                    "statut_du_recepisse",
+                    lambda x: "valide" if "Valide" in x.unique() else "invalide",
+                )
+            )
+            .reset_index()
+        )
+
+        df_spectacle[
+            ["siren", "statut_entrepreneur_spectacle", "est_entrepreneur_spectacle"]
+        ].to_csv(self.config.file_output, index=False)
+
+        ti = get_current_context()["ti"]
+        ti.xcom_push(
+            key=Notification.notification_xcom_key,
+            value=f"{str(df_spectacle.siren.nunique())} unités légales représentées.",
+        )
```

Comment:
There is a helper I added `DataProcessor._push_unique_count` that does this.

Reply:
What do you think about the placement of the Notification push ?

It's no big deal (again 😅 ) I just want to be consistent 👀 
I like it but  if the count is pushed here, and not after the file is sent to MinIO, the failure notification will include the count ?

Reply:
> There is a helper I added DataProcessor._push_unique_count that does this.

True! I will fix this

Reply:
>  I like it but if the count is pushed here, and not after the file is sent to MinIO, the failure notification will include the count ?

The notification should be pushed both for a failed or successful pipelines. It will look something like:
> 🔴 data_processing_entrepreneur_spectacle Données
> - preprocess_data(success): N rows were updated.
> - send_file_to_minio(failed)

Reply:
![Screenshot 2024-12-06 at 11 08 54](https://github.com/user-attachments/assets/46bc271e-efe6-4042-9cbf-8d3e17eaf101)
It looks like this for the other sources btw.

Reply:
I removed the task id because I didn't think it added much info to the notification.
I don't think that level of detail is needed in a Tchap alert. It should come from a detailed investigation.
What do you think ?

Reply:
It was in case we use an `all_done` trigger for example where a DAG would be green even though a particular task failed. This would go under the radar at the moment.
But if we never use it why not indeed!

Reply:
Hmm I see. I can't think of an existing use case in our workflows. But yes I see your point.
My only concern is oversharing on Tchap. But we can cercle back if need be of course.

Reply:
I checked the code you changed and we would still be notified for this particular use case, my bad, forget what I said. A Minio failure would look like:
> 🔴 data_processing_entrepreneur_spectacle: Données
> - N unités légales représentées..
> - send_file_to_minio(failed)

Reply:
Okay. i removed the `représentées` from the new notifications. I only used établissements ou unités légales. Less text.